src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
+        printHeader();
+
+        String input = "";
+        int i = 0;
+        int taskCount = 0;
+        Task[] tasks = new Task[100];
+
+        do {
+            input = getInput();
+
+            if (!input.equals("bye")) {
+                String[] splittedInput = input.split(" ");
+                if (input.equals("list")) {
+                    printListOutput(i, tasks);
+                } else if (splittedInput[0].equals("done")) {
+                    markTaskAsDone(tasks, splittedInput[1]);
+                } else {
+                    taskCount++;
+                    addNewTask(input, i, taskCount, tasks);
+                    i++;
+                }
+            }
+        } while (!input.equals("bye"));
+
+        printByeMessage();
+    }
+
+    private static String getInput() {
+        String input;
+        Scanner scanner = new Scanner(System.in);
+        input = scanner.nextLine();
+        printLineSeparator();
+        return input;
+    }
+
+    private static void printByeMessage() {
+        System.out.println("\tGoodbye! Hope to see you again soon!");
+        printLineSeparator();
+    }
+
+    private static void addNewTask(String input, int i, int taskCount, Task[] tasks) {
+        if (input.contains("deadline")) {
+            addDeadlineTask(input, i, tasks);
+        } else if (input.contains("event")) {
+            addEventTask(input, i, tasks);
+        } else {
+            addTodoTask(input, i, tasks);
+        }
+        System.out.println("\tAlright! I've just added this task:");
+        System.out.println("\t" + tasks[i].toString());
+        System.out.println("\tYou now have " + taskCount + " tasks on your task list.");
```

Comment:
Perhaps apply principle of SLAP more. Currently have varying levels of abstraction (System.out.println statements are now in the same level of abstraction as printLineSeparator().